lib/galaxy/tools/parameters/basic.py

```diff
@@ -1336,7 +1336,27 @@ def get_initial_value(self, trans, other_values):
     def get_legal_values(self, trans, other_values):
         if self.data_ref not in other_values:
             raise ValueError("Value for associated data reference not found (data_ref).")
-        return set(self.get_column_list(trans, other_values))
+        legal_values = self.get_column_list(trans, other_values)
```

Comment:
I don't think you should descend into HDCAs, that is not good for performance, and probably wrong as well (see line 1281 for how to handle this).

Reply:
Just pushed a commit with the suggested changes.